.github/workflows/CI.yml

```diff
@@ -112,6 +112,14 @@ jobs:
       needs.setup.outputs.workflows != 'false'
     uses: ./.github/workflows/eslint.yml
 
+  typecheck:
+    name: Typecheck
+    needs: [setup, prettier]
+    if: >-
+      needs.setup.outputs.frontend != 'false' ||
```

Comment:
Since this just checks types we could refine it further and only run it on *.ts *.tsx changes but I'm not if it's worth it as we might have to run it on every frontend change if we add a build step later as this would most likely be a requirement.